portality/lib/query_filters.py

```diff
@@ -13,13 +13,12 @@ def public_query_validator(q):
     if q.size() > 200:
         return False
 
-    # if the query has facets, that's fine
-    # otherwise, if it has no facets, only allow "count" style
-    # queries with zero results returned
-    if q.has_facets():
-        return True
-    else:
+    # we do not expect the fqw
+
+    if not q.has_facets():
         return q.size() == 0
+    else:
+        return True
```

Comment:
I don't understand this change - it looks like you've just reversed the if/else, which has the same overall effect?  If so, can we keep the original to avoid unnecessary file changes and to keep the original comment?  Or am I missing something?

Reply:
That is an error. Should be 
```
if q.has_facets()
  return q.size() == 0
else:
  return True
```
as the fqw should not have any facets.

Reply:
I'm still a bit confused by this change.  In the old logic it rejected a query if it did not have facets but did have a non-zero size parameter.  In the new logic it rejects a query if it has facets and has a non-zero size parameter.

The old logic would work against the fix query widget, but doesn't the new logic work against the public search?

I think what we actually want now is:

```
if q.has_facets():
    return q.size() > 0
else:
    return True
```

Which will just reject all queries which have a zero size and facets.